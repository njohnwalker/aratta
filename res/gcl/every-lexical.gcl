x := 10 ;
if x == y -> y := y + 1
[] x <= y -> y := 7 - y ;
do v > z -> do od
[] z < v -> if fi
od
[] foo >= baz | ~ true & 0 > 1 -> z := 4 + 30 - (x + y) + 24
[] true -> x,y,z := 10, 20, 76
fi ;
catdog := c
