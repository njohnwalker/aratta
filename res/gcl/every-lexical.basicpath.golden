Path 1
  [ Branch 1
      ; Assume false
      ; Substitute
          @x -> 10
      ; Assume x == y
      ; Substitute
          @y -> y + 1
      { HOLE }
    Branch 2
      ; Assume false
      ; Substitute
          @x -> 10
      ; Assume x <= y
      ; Substitute
          @y -> 7 - y
      { Satisfies INVARIANT == PLACEHOLDER }
    Branch 3
      ; Assume false
      ; Substitute
          @x -> 10
      ; Assume (foo >= baz | ~ true) & (0 > 1 | x % 2 == 0)
      ; Substitute
          @z -> (4 + 30) - ((x * y) / 24)
      { HOLE }
    Branch 4
      ; Assume false
      ; Substitute
          @x -> 10
      ; Assume true
      ; Substitute
          @x -> 10
          @y -> 20
          @z -> 76
      { HOLE }
  ]
  ; Substitute
      @catdog -> c
  { Satisfies true }
Path 2
  [ Branch 1
      ; Assume false
      ; Substitute
          @x -> 10
      ; Assume x == y
      ; Substitute
          @y -> y + 1
      { HOLE }
    Branch 2
      ; Assume INVARIANT == PLACEHOLDER
      { HOLE }
    Branch 3
      ; Assume false
      ; Substitute
          @x -> 10
      ; Assume (foo >= baz | ~ true) & (0 > 1 | x % 2 == 0)
      ; Substitute
          @z -> (4 + 30) - ((x * y) / 24)
      { HOLE }
    Branch 4
      ; Assume false
      ; Substitute
          @x -> 10
      ; Assume true
      ; Substitute
          @x -> 10
          @y -> 20
          @z -> 76
      { HOLE }
  ]
  ; Substitute
      @catdog -> c
  { Satisfies true }
Path 3
  [ Branch 1
      ; Assume false
      ; Substitute
          @x -> 10
      ; Assume x == y
      ; Substitute
          @y -> y + 1
      { HOLE }
    Branch 2
      ; Assume INVARIANT == PLACEHOLDER
      [ Branch 1
          ; Assume v > z
          { Satisfies quux >= ni }
        Branch 2
          [ 
          ]
          { HOLE }
      ]
      { Satisfies INVARIANT == PLACEHOLDER }
    Branch 3
      ; Assume false
      ; Substitute
          @x -> 10
      ; Assume (foo >= baz | ~ true) & (0 > 1 | x % 2 == 0)
      ; Substitute
          @z -> (4 + 30) - ((x * y) / 24)
      { HOLE }
    Branch 4
      ; Assume false
      ; Substitute
          @x -> 10
      ; Assume true
      ; Substitute
          @x -> 10
          @y -> 20
          @z -> 76
      { HOLE }
  ]
  ; Substitute
      @catdog -> c
  { Satisfies true }
Path 4
  [ Branch 1
      ; Assume false
      ; Substitute
          @x -> 10
      ; Assume x == y
      ; Substitute
          @y -> y + 1
      { HOLE }
    Branch 2
      ; Assume INVARIANT == PLACEHOLDER
      [ Branch 1
          ; Assume quux >= ni
          { HOLE }
        Branch 2
          [ 
          ]
          { HOLE }
      ]
      { Satisfies INVARIANT == PLACEHOLDER }
    Branch 3
      ; Assume false
      ; Substitute
          @x -> 10
      ; Assume (foo >= baz | ~ true) & (0 > 1 | x % 2 == 0)
      ; Substitute
          @z -> (4 + 30) - ((x * y) / 24)
      { HOLE }
    Branch 4
      ; Assume false
      ; Substitute
          @x -> 10
      ; Assume true
      ; Substitute
          @x -> 10
          @y -> 20
          @z -> 76
      { HOLE }
  ]
  ; Substitute
      @catdog -> c
  { Satisfies true }
Path 5
  [ Branch 1
      ; Assume false
      ; Substitute
          @x -> 10
      ; Assume x == y
      ; Substitute
          @y -> y + 1
      { HOLE }
    Branch 2
      ; Assume INVARIANT == PLACEHOLDER
      [ Branch 1
          ; Assume quux >= ni
          [ 
          ]
          { Satisfies quux >= ni }
        Branch 2
          [ 
          ]
          { HOLE }
      ]
      { Satisfies INVARIANT == PLACEHOLDER }
    Branch 3
      ; Assume false
      ; Substitute
          @x -> 10
      ; Assume (foo >= baz | ~ true) & (0 > 1 | x % 2 == 0)
      ; Substitute
          @z -> (4 + 30) - ((x * y) / 24)
      { HOLE }
    Branch 4
      ; Assume false
      ; Substitute
          @x -> 10
      ; Assume true
      ; Substitute
          @x -> 10
          @y -> 20
          @z -> 76
      { HOLE }
  ]
  ; Substitute
      @catdog -> c
  { Satisfies true }